File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    public int[] solution(int[] answers) {
        int[] cnt = new int[3];
        
        int[] a = new int[]{1,2,3,4,5};
        int[] b = new int[]{2,1,2,3,2,4,2,5};
        int[] c = new int[]{3,3,1,1,2,2,4,4,5,5};
        
        int one = 0;
        int two = 0;
        int three = 0;
        
        for (int i=0;i<answers.length;i++) {
            int answer = answers[i];
            if (a[i % a.length] == answer) one++;
            if (b[i % b.length] == answer) two++;
            if (c[i % c.length] == answer) three++;
            
        }
        
        int max_val = Math.max(one, Math.max(two, three));
        
        List<Integer> answer = new ArrayList<>();
        
        if (max_val == one) answer.add(1);
        if (max_val == two) answer.add(2);
        if (max_val == three) answer.add(3);
        
        
        
        
        
        
        
        return answer.stream().mapToInt(i -> i).toArray();
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    
    a = 0
    b = 0
    c = 0
    
    arr_a = [1,2,3,4,5]
    arr_b = [2, 1, 2, 3, 2, 4, 2, 5]
    arr_c = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    for i in range(len(answers)):
        if answers[i] == arr_a[i%len(arr_a)]:
            a += 1
        if answers[i] == arr_b[i%len(arr_b)]:
            b += 1
        if answers[i] == arr_c[i%len(arr_c)]:
            c += 1
    result = max(a,b,c)
    
    if a == result: answer.append(1)
    if b == result: answer.append(2)
    if c == result: answer.append(3)
    
    
    return answer